extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryFrameProcessorFactory.java

```diff
@@ -0,0 +1,181 @@
+  {
+    return query;
+  }
+
+  @JsonProperty("operatorList")
+  public List<OperatorFactory> getOperators()
+  {
+    return operatorList;
+  }
+
+  @JsonProperty("stageRowSignature")
+  public RowSignature getSignature()
+  {
+    return stageRowSignature;
+  }
+
+  @JsonProperty("emptyOver")
+  public boolean isEmptyOverFound()
+  {
+    return isEmptyOver;
+  }
+
+  @Override
+  public ProcessorsAndChannels<Object, Long> makeProcessors(
+      StageDefinition stageDefinition,
+      int workerNumber,
+      List<InputSlice> inputSlices,
+      InputSliceReader inputSliceReader,
+      @Nullable Object extra,
+      OutputChannelFactory outputChannelFactory,
+      FrameContext frameContext,
+      int maxOutstandingProcessors,
+      CounterTracker counters,
+      Consumer<Throwable> warningPublisher
+  )
+  {
+    // Expecting a single input slice from some prior stage.
+    final StageInputSlice slice = (StageInputSlice) Iterables.getOnlyElement(inputSlices);
+    final Int2ObjectSortedMap<OutputChannel> outputChannels = new Int2ObjectAVLTreeMap<>();
+
+    for (final ReadablePartition partition : slice.getPartitions()) {
+      outputChannels.computeIfAbsent(
+          partition.getPartitionNumber(),
+          i -> {
+            try {
+              return outputChannelFactory.openChannel(i);
+            }
+            catch (IOException e) {
+              throw new RuntimeException(e);
+            }
+          }
+      );
+    }
+
+    final Sequence<ReadableInput> readableInputs =
+        Sequences.simple(inputSliceReader.attach(0, slice, counters, warningPublisher));
+
+    final Sequence<FrameProcessor<Object>> processors = readableInputs.map(
+        readableInput -> {
+          final OutputChannel outputChannel =
+              outputChannels.get(readableInput.getStagePartition().getPartitionNumber());
+
+          return new WindowOperatorQueryFrameProcessor(
+              query,
+              readableInput.getChannel(),
+              outputChannel.getWritableChannel(),
+              stageDefinition.createFrameWriterFactory(outputChannel.getFrameMemoryAllocator()),
+              readableInput.getChannelFrameReader(),
+              frameContext.jsonMapper(),
+              operatorList,
+              stageRowSignature,
+              isEmptyOver
+          );
+        }
+    );
+
+    return new ProcessorsAndChannels<>(
+        ProcessorManagers.of(processors),
+        OutputChannels.wrapReadOnly(ImmutableList.copyOf(outputChannels.values()))
+    );
+  }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    WindowOperatorQueryFrameProcessorFactory that = (WindowOperatorQueryFrameProcessorFactory) o;
+    return Objects.equals(query, that.query)
+           && Objects.equals(operatorList, that.operatorList)
+           && Objects.equals(stageRowSignature, that.stageRowSignature);
```

Comment:
empty over is not there in both equals and hashcode.

Reply:
Addressed